### ISSUE TITLE
Padding transform to a multiple of 2

### ISSUE THREAD

**Issue by VinhLoiIT** [pytorch/vision#4865]
### 🚀 The feature

Introduce a new API to pad a tensor, such as `transforms.PadSizeDivisor` (I don't come up with a proper name at the moment so that it could be changed)

Expectation:
```
inputs = torch.rand(1,3,32,254)
tf = transforms.PadSizeDivisor(size_divisor=32, dim=3, pad_method='constant', pad_value=0)  # maybe we can support dim=[2,3]?
outputs = tf(inputs)
assert outputs.shape == torch.Size([1,3,32,256])
```

### Motivation, pitch

I think padding with the `size_divisor` argument is widely used in segmentation or U-like network where we apply downsampling followed by upsampling the input tensor. In most cases, the up/down ratio is multiple 2, e.g. 2, 4, 8, 16, 32. Therefore, the input is usually padded to some sort of these multiples such as 352, 768 in spatial dimensions (H, W).

The current implementation of `transforms.Pad` only supports fixed padding size. It works great during training where we usually apply the tensor crop or resize to a fixed size to the input tensor. However, during inference, we usually maintain the ratio when resizing. This leads to the remaining edge is not guaranteed to be a multiple of 2, and thus, the following up/down operators would be failed due to the size mismatch. At the moment, the solution is usually to compute the padding size on each edge before or after resizing the input image.

Therefore, I think it would be great to have a library-included transformation so that we can reuse it in our application.

**Comment by NicolasHug** [comment]
Thanks for the proposal @VinhLoiIT , I was actually looking into this today. I think this bit from the RAFT repo could be a starting point https://github.com/princeton-vl/RAFT/blob/aac9dd54726caf2cf81d8661b07663e220c5586d/core/utils/utils.py#L7


Instead of creating a new tranform however, I wonder if could be enough to provide a helper that returns the appropriate `pad` parameter to be passed to `torch.nn.functional.pad`

**Comment by datumbox** [comment]
FYI, **SOME** detection models (*RCNN) have something similar:
https://github.com/pytorch/vision/blob/d367a01a18a3ae6bee13d8be3b63fd6a581ea46f/torchvision/models/detection/transform.py#L227-L230
